content/what-keeps-us-awake-at-night-patching.md

```diff
@@ -0,0 +1,111 @@
+Title: What keeps us awake at night - Technical Debt
+Date: 2023-04-10
+Category: Posts 
+Tags: azure, azure-devops, eliminate-waste
+Slug: what-keeps-us-awake-at-night-patching
+Author: Willy-Peter Schaub
+Summary: Empower engineering without compromising security
+
+The obvious question that comes mind is: "**What is technical debt?**" I asked [ChatGPT](https://chat.openai.com/chat) and got the following response: "_Technical debt is a term used in software development to describe the additional work required to maintain and update a codebase developed using suboptimal practices. It can occur due to factors such as tight deadlines or the use of quick-and-dirty coding practices that prioritize speed over code quality and maintainability. Technical debt can result in higher long-term costs and slower development, limiting the ability of the software to adapt to changing business needs. Therefore, it is important to manage and minimize technical debt to reduce risks and ensure the sustainability of software development projects._"
+
+In a continued conversation another important nugget emerged from [ChatGPT](https://chat.openai.com/chat): "_Technical debt is typically considered to be self-imposed._"
+
+Personally, I agree with all of the above, but have a very specific visual that pops up in my mind whenever I am part of a discussion where technical debt raises its ugly head:
+
+> ![Technical Debt](../images/what-keeps-us-awake-at-night-patching-0.png)
+> Source [Jurassic Park](https://www.pastemagazine.com/movies/jurassic-park/the-12-best-quotes-from-jurassic-park)
+
+I am currently having sleepless nights with technical debt and vulnerabilities in two areas.
+
+---
+
+# Toxic server debt
+
+The first is the iron, the servers, that host the outcome of our software development. 
+
+> ![Cloud Accountability](../images/what-keeps-us-awake-at-night-patching-1.png)
+> Image Source: [Shared responsibility in the cloud](https://learn.microsoft.com/en-us/azure/security/fundamentals/shared-responsibility) 
+
+The image, I loaned from a Microsoft article, clearly shows that we are accountable for our servers and that we take responsibility to keep them up-to-date and secure. As we move to platform as a service (PaaS) or software as a service (SaaS) we delegate the accountability to the hosting vendor.
```

Comment:
Platform-as-a-Service (PaaS)
and 
Software-as-a-Service (SaaS)

I think there should be dashes in there.

Reply:
Correct, fixing it now.